### ISSUE TITLE
Foreground color of the REPL unchangeable

### ISSUE THREAD

**Issue by smee** [laurentpetit/ccw#834]
The black font used in the REPL can't be changed. The colors for evaluted content as well as errors can be configured at "Window -> Clojure -> Colors and Fonts -> (REPL Log - evaluated content | REPL Log - error).
The color used for standard output us illegible when using a dark theme in Eclipse (black on dark grey in this instance).
I tried all possibly relevant color settings for editors as well as under "General", to no avail.

**Comment by smee** [comment]
The problem seems to be related to the Ansi Console. When disabling the ANSI feature using the button on the REPL view the foreground color turns a readable grey.

**Comment by laurentpetit** [comment]
So the problem also occurs with the plain text editor?

Le mardi 4 août 2015, Steffen Dienst <notifications@github.com> a écrit :

> The problem seems to be related to the Ansi Console. When disabling the
> ANSI feature using the button on the REPL view the foreground color turns a
> readable grey.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/issues/834#issuecomment-127662076>.
>


-- 
Laurent Petit

**Comment by smee** [comment]
No, the editors are fine. The problem seems to be related to the use of the ANSI console plugin. When enabled it outputs all text (for example standard output from the running clojure REPL) in black. When I disable it using the button on the top right of the REPL view, I see a more friendly color. 
I think this means the problem is not within CCW but rather due to the fact, that all output without any escape codes gets rendered as black when ANSI console is enabled. So I suppose I should file an issue at this plugin's tracker to respect the default colors set in eclipse when rendering black? If so, this ticket here should be closed, right?

**Comment by laurentpetit** [comment]
Two thoughts:
- there's a preference page for the ansi console. Maybe you can find a work
around by choosing a different color from black ?
- I also think this deserves an issue in ansi console tracker. But please
keep this issue open, and cross reference the issues. So that when it's
fixed in ansicobsole we can also react and deliver new ccw releases with
it.

Le mercredi 5 août 2015, Steffen Dienst <notifications@github.com> a écrit :

> No, the editors are fine. The problem seems to be related to the use of
> the ANSI console plugin. When enabled it outputs all text (for example
> standard output from the running clojure REPL) in black. When I disable it
> using the button on the top right of the REPL view, I see a more friendly
> color.
> I think this means the problem is not within CCW but rather due to the
> fact, that all output without any escape codes gets rendered as black when
> ANSI console is enabled. So I suppose I should file an issue at this
> plugin's tracker to respect the default colors set in eclipse when
> rendering black? If so, this ticket here should be closed, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/issues/834#issuecomment-127888514>.
>


-- 
Laurent Petit

**Comment by smee** [comment]
The configuration options of the ANSI console only allow changes to the pallette used for its colors, but not to the default color if no ANSI escape codes are in effect. I filed an issue over here: https://github.com/mihnita/ansi-econsole/issues/15

**Comment by laurentpetit** [comment]
Hmm, maybe it's a bug in CCW, after all. Doing some tests right now, and I can see that the problem does not occur in the Console with a simple java program outputting ansi codes

**Comment by laurentpetit** [comment]
No, it was, to my opinion, a bug in AnsiEConsole which tries to control too much the foreground color. I submitted a patch.

**Comment by laurentpetit** [comment]
Moving this issue to 0.34.0 since I will release 0.33.0 very soon

**Comment by arichiardi** [comment]
Good job, thanks Laurent.

**Comment by mihnita** [comment]
Hi all,

I am the author of the AnsiConsole, and this is a ping regarding issue #15 (https://github.com/mihnita/ansi-econsole/issues/15)

I have added a comment there explaining why I think this is actually not a bug in AnsiConsole.

The gist of it is that there are two sets of background / foreground colors, the editor ones and the Eclipse Console ones. They can be configured independently, and AnsiConsole honors the Eclipse Console ones.

You can read my comment on August 9 for a more complete description.
I just wanted to make sure that you are not waiting for that to be fixed.
But I am open to suggestions and working together for find a solution that works for everybody.

Thank you,
Mihai
